ledfx/utilities/gradient_extraction.py

```
@@ -0,0 +1,989 @@
+    """
+    start_time = time.time()
+
+    # Handle both path and PIL Image inputs
+    if isinstance(image_source, str):
+        # Path provided - open and manage the image
+        try:
+            with Image.open(image_source) as pil_image:
+                return _extract_gradient_metadata_from_image(
+                    pil_image, start_time
+                )
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)